lib/errors.js

```diff
@@ -48,11 +48,40 @@ class FileError extends Error {
   }
 }
 
+class DeserializationError extends Error {
+  constructor(message, contentType, err) {
+    super(message);
+    this.contentType = contentType;
+    this.name = this.constructor.name;
+    for (let e in err) this[e] = err[e];
+  }
+}
+
+class SerializationError extends Error {
+  constructor(message, contentType, err) {
+    super(message);
+    this.contentType = contentType;
+    this.name = this.constructor.name;
+    for (let e in err) this[e] = err[e];
```

Comment:
why?

Reply:
I actually don't know. I saw one of the other errors doing it and figured it was being done for a reason.